lenstronomy/Sampling/parameters.py

```
@@ -544,24 +591,41 @@ def update_lens_scaling(self, kwargs_special, kwargs_lens, inverse=False):
         :return: updated lens model keyword argument list
         """
         kwargs_lens_updated = copy.deepcopy(kwargs_lens)
-        if self._mass_scaling is False:
+        # If we do not scaling, there's nothing to be done
+        if not (self._mass_scaling or self._general_scaling):
             return kwargs_lens_updated
-        scale_factor_list = np.array(kwargs_special['scale_factor'])
-        if inverse is True:
-            scale_factor_list = 1. / np.array(kwargs_special['scale_factor'])
-        for i, kwargs in enumerate(kwargs_lens_updated):
-            if self._mass_scaling_list[i] is not False:
-                scale_factor = scale_factor_list[self._mass_scaling_list[i] - 1]
-                if 'theta_E' in kwargs:
-                    kwargs['theta_E'] *= scale_factor
-                elif 'alpha_Rs' in kwargs:
-                    kwargs['alpha_Rs'] *= scale_factor
-                elif 'alpha_1' in kwargs:
-                    kwargs['alpha_1'] *= scale_factor
-                elif 'sigma0' in kwargs:
-                    kwargs['sigma0'] *= scale_factor
-                elif 'k_eff' in kwargs:
-                    kwargs['k_eff'] *= scale_factor
+
+        if self._mass_scaling:
```

we could think of even fully removing mass_scaling and replace it with something within the power_law scaling that automatically sets the power=1. You don't need to do it, but perhaps mark a TODO item in the code, or even a DeprecationWarning that this will come.

---

that makes a lot of sense, I thought about having the general_scaling default the power law index to 1 as well. I decided not to because it could cause confusion, if for instance someone provides the power law index with a typo and lenstronomy silently defaults to 1. I think reimplementing mass_scaling to use the same machinery as general_scaling makes most sense. I am adding a TODO along these lines

---

ok, we can leave it like the and thanks for marking it as TODO